vignettes/internal-impact.Rmd

```diff
@@ -0,0 +1,274 @@
+  burden_outcome = "hepb_deaths_acute,hepb_deaths_dec_cirrh,hepb_deaths_hcc;hepb_cases_acute_severe,hepb_cases_dec_cirrh,hepb_cases_hcc")
+t <- tempfile(fileext = ".csv")
+write.csv(recipe, t, row.names = FALSE)
+```
+
+This is a set of properties defining what data we want to extract from the db e.g. `touchstone`, `modelling_group`, `disease`, focal and baseline scenarios and `burden_outcome`. It captures the same info as the args to `calculate_impact` above. Then use the recipe to define meta data frame
+
+
+```r
+meta <- vimpact:::get_meta_from_recipe(default_recipe = FALSE, recipe = t, con = con)
+```
+
+And use this to calculate impact
+
+
+```r
+old_impact <- vimpact:::get_raw_impact_details(con, meta, "deaths")
+str(old_impact)
+#> 'data.frame':	9898 obs. of  7 variables:
+#>  $ country       : int  104 104 104 104 104 104 104 104 104 104 ...
+#>  $ time          : int  2000 2001 2002 2003 2004 2005 2006 2007 2008 2009 ...
+#>  $ baseline_value: num  5636 5765 5901 6034 6172 ...
+#>  $ focal_value   : num  5636 5765 5901 6015 6075 ...
+#>  $ value         : num  0 0 0 19 97 180 252 312 344 381 ...
+#>  $ index         : int  1 1 1 1 1 1 1 1 1 1 ...
+#>  $ burden_outcome: chr  "deaths" "deaths" "deaths" "deaths" ...
+```
+
+## Impact by year of vaccination
+
+### Function interface
+
+Very similar to examples for calendar year and birth year, to calculate impact by year of vaccination stratified by activity type run
+
+
+```r
+impact <- vimpact::calculate_impact(
+  con, method = "yov_activity_type", touchstone = "201910gavi-5",
+  modelling_group = "CDA-Razavi",  disease = "HepB",
+  focal_scenario_type = "default", focal_vaccine_delivery = list(
+    list(vaccine = "HepB_BD", activity_type = "routine"),
+    list(vaccine = "HepB", activity_type = "routine")
+  ),
+  baseline_scenario_type = "novac",
+  burden_outcomes = "dalys")
+str(impact)
+#> tibble [4,432 × 6] (S3: tbl_df/tbl/data.frame)
+#>  $ country       : chr [1:4432] "AFG" "AFG" "AFG" "AFG" ...
+#>  $ vaccine       : chr [1:4432] "HepB" "HepB" "HepB" "HepB" ...
+#>  $ activity_type : chr [1:4432] "routine" "routine" "routine" "routine" ...
+#>  $ year          : int [1:4432] 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 ...
+#>  $ burden_outcome: chr [1:4432] "dalys" "dalys" "dalys" "dalys" ...
+#>  $ impact        : num [1:4432] 91667 94065 93427 98811 101646 ...
+```
+
+or use `method = "yov_birth_cohort` for impact by year of vaccination stratified by birth cohort. This uses the same logic as other impact methods but before calling the public facing impact function it will also extract FVP data from the data base for this `touchstone` and vaccine delivery. It then calls either `impact_by_year_of_vaccination_activity_type` or `impact_by_year_of_vaccination_birth_cohort` to get impact.
+
+### Recipe interface
+
+We can re-use the recipe from above and get meta table for this method
+
+
+```r
+meta <- vimpact:::get_meta_from_recipe(default_recipe = FALSE, recipe = t,
```

Comment:
should we be advocating using private access here? `:::` used in a few places

Reply:
Probably not but this is one of Xiang's functions which will be removed eventually